### ISSUE TITLE
Make `llima-compile` automatically select Gemma 3 sliding-attention cache models

### ISSUE THREAD

**Issue by ashish-negi-sima-ai** [sima-neat/llima#68]
## Problem

Gemma 3 models use sliding-window attention / hybrid cache behavior. Compiling them correctly currently requires model-specific compile logic to select the right cache models and language group offsets.

`llima-compile` should infer this automatically from the model config, instead of requiring users to pass custom Gemma-specific arguments or wrapper scripts.

## Examples

Gemma 3 4B uses a `1024` sliding window:

```bash
LANGUAGE_GROUP_SIZE=128
LANGUAGE_GROUP_OFFSETS=$(seq -s ',' 0 ${LANGUAGE_GROUP_SIZE} $((MAX_NUM_TOKENS - 1))),$((1024 - LANGUAGE_GROUP_SIZE))

llima-compile \
    -c ${TMPDIR}/config.py \
    --language_group_size ${LANGUAGE_GROUP_SIZE} \
    --language_group_offsets ${LANGUAGE_GROUP_OFFSETS} \
    --max_num_tokens ${MAX_NUM_TOKENS} \
    $@
```

Gemma 3 1B uses a `512` sliding window:

```bash
LANGUAGE_GROUP_SIZE=128
LANGUAGE_GROUP_OFFSETS=$(seq -s ',' 0 ${LANGUAGE_GROUP_SIZE} $((MAX_NUM_TOKENS - 1))),$((512 - LANGUAGE_GROUP_SIZE))

llima-compile \
    -c ${TMPDIR}/config.py \
    --language_group_size ${LANGUAGE_GROUP_SIZE} \
    --language_group_offsets ${LANGUAGE_GROUP_OFFSETS} \
    --max_num_tokens ${MAX_NUM_TOKENS} \
    $@
```

## Requested Behavior

For Gemma 3 models, `llima-compile` should:

- Detect `model_type: gemma3` / `gemma3_text`.
- Read the model's `sliding_window` value.
- Automatically choose the needed cache models for sliding attention.
- Automatically derive `language_group_offsets`.
- Work without a Gemma-specific compile wrapper.

Suggested default:

```text
language_group_size = 128
language_group_offsets = seq(0, language_group_size, max_num_tokens - 1)
                         plus (sliding_window - language_group_size)
```

So:

```text
Gemma 3 1B: 512 - 128 = 384
Gemma 3 4B: 1024 - 128 = 896
```

## Acceptance Criteria

- `llima-compile google/gemma-3-1b-it ...` works without a custom script.
- `llima-compile google/gemma-3-4b-it ...` works without a custom script.
- Required sliding-attention cache models are created/selected automatically.
- Generated artifacts run on Modalix for short and long prompts.
- `--return_logits` compilations also work for accuracy benchmarking.
- The resolved cache/sliding-window settings are recorded in compile metadata for reproducibility.
